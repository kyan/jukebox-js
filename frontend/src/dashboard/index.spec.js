import React from 'react'
import { shallow } from 'enzyme'
import Dashboard from './index'

describe('Dashboard', () => {
  const onPlayMock = jest.fn().mockName('onPlayMock')
  const onStopMock = jest.fn().mockName('onStopMock')
  const onPauseMock = jest.fn().mockName('onPauseMock')
  const onNextMock = jest.fn().mockName('onNextMock')
  const onPreviousMock = jest.fn().mockName('onPreviousMock')
  const onVolumeChangeMock = jest.fn().mockName('onVolumeChangeMock')
  const onDropMock = jest.fn().mockName('onDropMock')
  const onTracklistClearMock = jest.fn().mockName('onTracklistClearMock')
  const onRemoveTrackMock = jest.fn().mockName('onRemoveTrackMock')
  const onSearchClickMock = jest.fn().mockName('onSearchClickMock')

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    it('renders the happy path correctly', () => {
      const props = {
        online: true,
        disabled: false,
        volume: 23,
        playbackState: 'playing',
        onPlay: onPlayMock,
        onStop: onStopMock,
        onPause: onPauseMock,
        onNext: onNextMock,
        onPrevious: onPreviousMock,
        onVolumeChange: onVolumeChangeMock,
        onDrop: onDropMock,
        onTracklistClear: onTracklistClearMock,
        onRemoveTrack: onRemoveTrackMock,
        onSearchClick: onSearchClickMock,
        trackListImages: {},
        tracklist: ['track1', 'track2'],
        currentTrack: { title: 'track1' }
      }
      const wrapper = shallow(<Dashboard {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
