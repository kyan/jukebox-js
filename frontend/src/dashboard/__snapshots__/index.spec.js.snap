// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render renders the happy path correctly 1`] = `
<DimmerDimmable
  blurring={true}
  dimmed={false}
>
  <DragInTrack
    disabled={false}
    onDrop={[MockFunction onDropMock]}
  >
    <SearchContainer>
      <div
        className="c-header"
      >
        <Controls
          disabled={false}
          onNext={[MockFunction onNextMock]}
          onPause={[MockFunction onPauseMock]}
          onPlay={[MockFunction onPlayMock]}
          onPrevious={[MockFunction onPreviousMock]}
          onStop={[MockFunction onStopMock]}
        />
        <div
          className="c-header__actions"
        >
          <SearchButton
            disabled={false}
            onClick={[MockFunction onSearchClickMock]}
          />
          <VolumeButtons
            disabled={false}
            onVolumeChange={[MockFunction onVolumeChangeMock]}
          />
          <Settings />
        </div>
      </div>
      <div
        className="c-main"
      >
        <div
          className="c-main__nowPlaying"
        >
          <CurrentTrackContainer />
        </div>
        <div
          className="c-main__trackList"
        >
          <h6>
            Tracklist
            <ClearPlaylist
              disabled={false}
              onClear={[MockFunction onTracklistClearMock]}
            />
          </h6>
          <Tracklist
            currentTrack={
              Object {
                "title": "track1",
              }
            }
            disabled={false}
            onArtistSearch={[MockFunction onArtistSearch]}
            onRemoveTrack={[MockFunction onRemoveTrackMock]}
            tracks={
              Array [
                "track1",
                "track2",
              ]
            }
          />
        </div>
      </div>
    </SearchContainer>
  </DragInTrack>
</DimmerDimmable>
`;
